jenkins/download-test.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+# Pass `NODE_VERSION` as an environment variable. Can be an exact or vague version.
+# NODE_VERSION=v8 ./download-test.sh --tap ./test.tap
+
+while [ $# -gt 1 ]; do
+
+  case "$1y" in
+      -t|--tap)
+      tapFile="$2"
+      shift # Past option argument.
+      ;;
+      *) # unknown option
+      echo "Ignoring unknown option: $1"
+      ;;
+  esac
+  shift # Past option.0
+done
+
+if [ "$DOWNLOAD_LOCATION" == "Nightly" ]; then
+  downloadDir="nightly"
+else
+  downloadDir="release"
+fi
+
+if [ -z "$NODE_VERSION" ]; then
+  echo "NODE_VERSION is undefined! Please declare NODE_VERSION"
+  exit 1
+fi
+
+# If the provided NODE_VERSION is inexact, e.g. `v8`, get the most recent matching version.
+LINK=`rsync rsync://unencrypted.nodejs.org/nodejs/$downloadDir/ | grep $NODE_VERSION | sort -t. -k 1,1n -k 2,2n -k 3,3n | tail -1 | awk '{print $5}'`
+
+# Remove old files
+[ "$tapFile" ] && rm -rf "$tapFile" && echo "TAP version 13" > $tapFile
+rm -rf v*
```

Comment:
move this up a line, otherwise if people call their tapfile vsomething they're going to be very confused.